### PR TITLE
Module 2

### PR DESCRIPTION
**Pull request by Jonesey13**
@CharlieCumber1 Please can you create the PR for future project exercise reviews?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -114,3 +114,7 @@ env/
 
 ### Configuration files ###
 .env
+
+
+# Trello API config
+trello_config.py
```
Comment (Jonesey13) [inline .gitignore:120]: Should probably document the use of this file in the README

File: app.py [hunk 2]
```diff
@@ -1,32 +1,25 @@
 from flask import Flask, render_template, request, redirect, url_for
 from operator import itemgetter 
-import session_items as session
-
+import trello_items as trello
 
 app = Flask(__name__)
 app.config.from_object('flask_config.Config')
```
Comment (Jonesey13) [inline app.py:6]: This line (and the `flask_config.py` file) is no longer needed

**Comment by Jonesey13** [comment]
Incorrect file name (`requirments.txt`), which has also been setup as a binary file in git (this is happening for alot of people in this cohort; a PyCharm IDE bug or something?)

This also breaks the setup of the project as the `requests` package is not installed when running `source setup.sh`.

File: trello_items.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import requests
+from trello_config import KEY, TOKEN, BOARD, PENDING, DONE
```
Comment (Jonesey13) [inline trello_items.py:2]: PENDING & DONE could be inferred from the list names (technically the board name could be as well)

File: trello_items.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import json
+from flask import session
```
Comment (Jonesey13) [inline trello_items.py:4]: unused import

File: trello_items.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        if card['idList'] == PENDING:
+            self.status = 'Not Started'
+        elif card['idList'] == DONE:
+            self.status = 'Completed'
+        if card['desc'] != 'None':
+            self.description = card['desc']
+
+
+def get_items():
+    response = requests.get('https://api.trello.com/1/boards/{id}/cards?key={key}&token={token}'.format(id=BOARD, key=KEY, token=TOKEN))
```
Comment (Jonesey13) [inline trello_items.py:19]: Should probably use the `params` option here rather than use a query string.

File: trello_items.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    all_items = [Item(item) for item in response.json()]
+    return all_items
+
+def get_item(id):
+    items = get_items()
+    return next((item for item in items if item.id == id), None)
```
Comment (Jonesey13) [inline trello_items.py:25]: Good use a generator comprehension!

File: trello_items.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+def add_item(title, description):
+    requests.post('https://api.trello.com/1/cards?key={key}&token={token}&idList={listId}&name={name}&desc={desc}'.format(key=KEY, token=TOKEN, listId=PENDING, name=title, desc=description))
```
Comment (Jonesey13) [inline trello_items.py:28]: Probably easier to use a format string:
```python
f'https://api.trello.com/1/cards?key={KEY}&token={TOKEN}&idList={PENDING}&name={title}&desc={description}'
```

File: trello_items.py (hunk 3 above)
Comment (CharlieCumber1) [reply trello_items.py:2]: I am not quite sure how I can eliminate the need for these variables as they are used across multiple functions.

Comment (CharlieCumber1) [reply trello_items.py:2]: I have made the updates on my module 3 branch which will be included in the next PR

**Pull request merged**